core-tests/src/test/java/org/mule/runtime/core/processor/strategy/ReactorProcessingStrategyTestCase.java

```diff
@@ -151,14 +151,27 @@ public void tx() throws Exception {
   }
 
   @Override
-  @Description("When the ReactorProcessingStrategy is configured and a transaction is active processing fails with an error")
+  @Description("When the ReactorProcessingStrategy is configured any async processing will be returned to CPU_LIGHT thread. "
+      + "This helps avoid deadlocks when there are reduced number of threads used by async processor.")
   public void asyncCpuLight() throws Exception {
     super.asyncCpuLight();
-    assertThat(threads, hasSize(2));
-    assertThat(threads.stream().filter(name -> name.startsWith(CPU_LIGHT)).count(), equalTo(1l));
+    assertThat(threads.size(), between(1, 2));
+    assertThat(threads.stream().filter(name -> name.startsWith(CPU_LIGHT)).count(), between(1l, 2l));
+    assertThat(threads.stream().filter(name -> name.startsWith(IO)).count(), equalTo(0l));
+    assertThat(threads.stream().filter(name -> name.startsWith(CPU_INTENSIVE)).count(), equalTo(0l));
+    assertThat(threads.stream().filter(name -> name.startsWith(CUSTOM)).count(), equalTo(0l));
+  }
+
+  @Override
+  @Description("When the ReactorProcessingStrategy is configured any async processing will be returned to CPU_LIGHT thread. "
+      + "This helps avoid deadlocks when there are reduced number of threads used by async processor.")
+  public void asyncCpuLightConcurrent() throws Exception {
+    super.asyncCpuLightConcurrent();
+    assertThat(threads.size(), between(2, 3));
+    assertThat(threads.stream().filter(name -> name.startsWith(CPU_LIGHT)).count(), between(2l, 3l));
     assertThat(threads.stream().filter(name -> name.startsWith(IO)).count(), equalTo(0l));
     assertThat(threads.stream().filter(name -> name.startsWith(CPU_INTENSIVE)).count(), equalTo(0l));
-    assertThat(threads.stream().filter(name -> name.startsWith(CUSTOM)).count(), equalTo(1l));
+    assertThat(threads.stream().filter(name -> name.startsWith(CUSTOM)).count(), equalTo(0l));
```

Comment:
there are already many assertions like this, but you may use `not(IsCollectionContaining.hasItem(Matchers.startsWith(CUSTOM)))` instead of doing the stream/filter thing